2023/incoming/my-top-7-new-features.pod

```diff
@@ -0,0 +1,182 @@
+	}
+
+	do_it( 'brian', 3, 'NYC' ); # limit is the default
+
+	do_it( 'brian', 3, undef, 7 ); # where is the default
+
+Eventually, signatures will turn off C<@_> inside the braces, which has been annoying, but if that's the trade-off for signatures, so be it.
+
+=head2 1. Substitution returns a copy.
+
+Perl v5.14 added the L<the r flag for the substitution operator|https://www.effectiveperlprogramming.com/2010/09/use-the-r-substitution-flag-to-work-on-a-copy/>. When I use that, the bound string is unmodified but I get the modified version of the
+
+	my $modified = $original =~ s/PATTERN/REPLACEMENT/r;
+
+This is especially handy in a C<map>, where I have often made the mistake of transforming C<@original> into a list of C<1>s or empty strings:
+
+	my @modified = map { s/PATTERN/REPLACEMENT/r } @original;
+
+=head2 2. Postfix dereference
+
+Perl's generalized dereferencing is a circumfix operatar, which is a fancy way of saying that I have to type on both sides of the reference:
+
+	@{ REFERENCE }
+	@{ $hash{$key} }
+
+	foreach my $item ( @{ $hash{$key} } ) {
+		...
+	}
+
+Perl v5.20 added the L<postfix dereference feature|https://www.effectiveperlprogramming.com/2014/09/use-postfix-dereferencing/>, and Perl v5.24 stabilized it:
+
+	use v5.24;
+
+	foreach my $item ( $hash{$key}->@* ) {
+		...
+	}
+
+=head2 3. The infix isa
+
+Checking that a Perl object was of a particular class, or a subclass or that, was a pain. There's C<isa> from L<UNIVERSAL>, but that only works if you have an object. I do this quite a bit in testing Perl code, and since it might blow up, I have to wrap that in an C<eval>:
+
+    if( eval { $obj->isa($target_class) } ) {
+    	...
+    }
+
+Perl v5.32 introduced the L<infix version of isa|https://www.effectiveperlprogramming.com/2020/01/use-the-infix-class-instance-operator/> that can return false if the left-hand side is not an object. This is much more pleasing:
+
+	if( $obj isa $target_class ) {
+		...
+	}
+
+Someday we might get infix operators for C<can> and C<does>.
+
+=head2 4. Key-value slices
+
+I'm a big fan of slices, but for a long time (since forever), Perl didn't have a way to slice a hash and get a hash back. For example, I want a sub-hash that I can use a C<map>. I've typed out something like this so often I don't even feel the pain anymore:
+
+	my %slice = map { $_ => $hash{$_} } @wanted_keys;
+
+Perl v5.20 introduced L<"key-value" slices|https://www.effectiveperlprogramming.com/2014/07/perl-5-20-introduces-keyvalue-slices/>. I know it's a key-value slice because I see the C<%> in front of C<hash>:
+
+	use v5.20;
+	my %key_value_slice = %hash{@wanted_keys};
+
+A hash slice with C<@> gives you back only the values:
+
+	my %value_slice = @hash{@wanted_keys};
+
+In the last year I'm been particularly keen to clean up areas where I'm passing around too much data. Instead of using a "god" hash with lots of irrelevant keys, I extract just the parts I need to nothing gets more than they should be allowed to use:
```

Comment:
```suggestion
In the last year I'm been particularly keen to clean up areas where I'm passing around too much data. Instead of using a "god" hash with lots of irrelevant keys, I extract just the parts I need so that nothing gets more than it should be allowed to use:
```